File: haskell/backup.hs
{-
comenzarJuego :: [Jugador] -> [Carta] -> [Carta] -> Int -> IO [Jugador]
comenzarJuego jugadores mesa mazo num
    | null mazo && manosVacias jugadores = return jugadores
    | manosVacias jugadores = do
        comenzarJuego (servirJugadores jugadores mazo)
            mesa (drop (length jugadores * 3) mazo) num
    | otherwise = do
        let turno = (jugadores !! num, mesa)

        putStrLn ""
        putStrLn "--------------------------------------------------------------------"
        print $ "Jugador: " ++ show (fst turno)
        print $ "Mesa: " ++ show (snd turno)

        
        let turnoTerminado = jugar turno
        turnoTerminado1 <- turnoTerminado

        let player = fst turnoTerminado1
            tapete = snd turnoTerminado1
            jugadoresActualizados = reemplazarJugador player jugadores num
            ronda = (num + 1) `mod` length jugadores

        let test = (Jugador (nombre player) [] (monton player), snd turno)
            test2 = reemplazarJugador (fst test) jugadoresActualizados num
        comenzarJuego test2 tapete mazo ronda

        {- comenzarJuego jugadoresActualizados tapete mazo ronda -}
-}

{-
jugar :: (Jugador, [Carta]) -> IO (Jugador, [Carta])
jugar turno = do
    putStr "Escoba: (S)i - (N)o: "
    respuesta <- getLine
    putStr "Elija una carta de su mano: "
    cartaMano <- getLine
    let cartaManoElegida = (read cartaMano :: Int) - 1
    jugada turno (escoba respuesta) cartaManoElegida
-}

{-
escoba :: String -> Bool
escoba respuesta = respuesta == "S"
-}

{-
jugada :: (Jugador, [Carta]) -> Bool -> Int -> IO (Jugador, [Carta])
jugada (jugador, mesa) escoba posicionCarta
    | not escoba = return (Jugador (nombre jugador) (filter (/= head cartasQuitar) (cartas jugador)) (monton jugador), mesa ++
    [cartas jugador !! posicionCarta])
    | otherwise = do
        putStr "Cantidad cartas en mesa a usar: "
        cantidadCartasMesa <- getLine
        cartasMesaElegidas <- elegirCartasMesa (read cantidadCartasMesa :: Int) mesa []
        let listaCartas = (cartas jugador !! posicionCarta) : cartasMesaElegidas
        return $ actualizarJugada listaCartas (jugador, mesa) (verificarEscoba listaCartas)
-}

{-
actualizarJugada :: [Carta] -> (Jugador, [Carta]) -> Bool -> (Jugador, [Carta])
actualizarJugada cartasQuitar (jugador, mesa) escoba
    | escoba = (Jugador (nombre jugador) (filter (/= head cartasQuitar) (cartas jugador)) (monton jugador ++ cartasQuitar), 
                filter (`notElem`  tail cartasQuitar) mesa)
    | otherwise = (Jugador (nombre jugador) (filter (/= head cartasQuitar) (cartas jugador)) (monton jugador), 
                    mesa ++ [head cartasQuitar])
-}

----------------------------------------------------------------------------------------------------------------

{-
quitarCarta :: [Carta] -> Int -> [Carta]
quitarCarta cartas posicion
  | null cartas = []
  | posicion == 0 = quitarCarta (tail cartas) (posicion-1)
  | otherwise = head cartas : quitarCarta (tail cartas) (posicion-1)
-}

File: python/backup.py
"""
    turno = []

    for jugador in jugadores:
        turno.append((jugador, mesa))
"""

"""def elegirCartasMesa(mesa):
    print("Mesa: ", mesa)
    cartas_elegidas = []
    cantidad_cartas_mesa = len(mesa)
    cantidad_cartas_mesa_usar = int(input('Cantidad cartas en mesa a usar: (1-%d): ' % cantidad_cartas_mesa))
    for x in range(cantidad_cartas_mesa_usar):
        carta_elegida = 0
        while carta_elegida < 1 or carta_elegida > cantidad_cartas_mesa:
            carta_elegida = int(
                input('Elija la/s carta/s de la mesa que suma/n 15 con la de su mano: (1-%d): ' % cantidad_cartas_mesa))
        carta_mesa = mesa[carta_elegida - 1]
        cartas_elegidas.append(carta_mesa)
    return cartas_elegidas

def jugar(turno):
    jugador = turno[0]
    mesa = turno[1]

    estadoTurno(jugador, mesa)

    escoba = input('Escoba: (S)i - (N)o: ').upper()
    if escoba == 'S':
        carta_mano_elegida = elegirCartaMano(jugador)
        cartas_mesa_elegidas = elegirCartasMesa(mesa)
        if not verificarEscoba(carta_mano_elegida, cartas_mesa_elegidas):
            print("EL JUGADOR SE HA EQUIVOCADO - Debera descartar una carta")
            descartarCarta(jugador, mesa)
        else:
            actualizarMonton(jugador, mesa, carta_mano_elegida,cartas_mesa_elegidas)
    else:
        print("Debera descartar una carta")
        descartarCarta(jugador, mesa)"""